test/integration/targets/podman_container/tasks/main.yml

```diff
@@ -3,6 +3,16 @@
     - ansible_facts.virtualization_type != 'docker'
     - ansible_facts.distribution in ['RedHat', 'Fedora']
   block:
+
+    - name: Delete all container leftovers from tests if exist
```

Comment:
I don't think that you should do this here. It shadows possible problems existing before the test run. Only clean up resources you created, not something random.